memory_test.go

```diff
@@ -0,0 +1,162 @@
+package gosql
+
+import (
+	"errors"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var mb *MemoryBackend
+
+func init() {
+	helpMessagesDisabled = true
```

Comment:
I'd pass this into Parse and maybe have it keep some internal state, this is going to be necessary at some point in general anyway. I'll push a change to this branch